File: php/program/BarangElektronik.php
<?php
// Class Level 1: BarangElektronik
class BarangElektronik {
    private $id;
    private $nama;
    private $merek;
    private $harga;
    private $fotoProduk;

    public function __construct($id, $nama, $merek, $harga, $fotoProduk) {
        $this->id = $id;
        $this->nama = $nama;
        $this->merek = $merek;
        $this->harga = $harga;
        $this->fotoProduk = $fotoProduk;
    }

    // Getters
    public function getId() { return $this->id; }
    public function getNama() { return $this->nama; }
    public function getMerek() { return $this->merek; }
    public function getHarga() { return $this->harga; }
    public function getFotoProduk() { return $this->fotoProduk; }
    
    // Setters
    public function setNama($nama) { $this->nama = $nama; }
    public function setMerek($merek) { $this->merek = $merek; }
    public function setHarga($harga) { $this->harga = $harga; }
    public function setFotoProduk($path) { $this->fotoProduk = $path; }

    public function __destruct() {
    }
}
?>

File: cpp/program/BarangElektronik.cpp
// Class Level 1: BarangElektronik
class BarangElektronik {
protected:
    int id;
    string nama;
    string merek;
    double harga;

public:
    BarangElektronik(int id, string nama, string merek, double harga)
        : id(id), nama(nama), merek(merek), harga(harga) {}
    
    virtual ~BarangElektronik() {
    }

    int getId() const { return id; }
    string getNama() const { return nama; }
    string getMerek() const { return merek; }
    double getHarga() const { return harga; }

    void setNama(const string& nama) { this->nama = nama; }
    void setMerek(const string& merek) { this->merek = merek; }
    void setHarga(double harga) { this->harga = harga; }

    virtual void display(const map<string, int>& colWidths) const {
        cout << "|" << setw(colWidths.at("ID")) << left << getId() << " | "
             << setw(colWidths.at("Nama")) << left << getNama() << " | "
             << setw(colWidths.at("Merek")) << left << getMerek() << " | "
             << setw(colWidths.at("Harga")) << left << fixed << setprecision(2) << getHarga() << " | ";
    }
};
